Modules/mmapmodule.c

```diff
@@ -725,7 +723,7 @@ mmap_size_method(PyObject *op, PyObject *Py_UNUSED(ignored))
     CHECK_VALID(NULL);
 
 #ifdef MS_WINDOWS
-    if (self->file_handle != INVALID_HANDLE_VALUE) {
+    if (self->file_handle != INVALID_HANDLE_VALUE || !self->trackfd) {
```

Comment:
I don't like combining an error condition check with a state check like this, it gives us a way to call `GetFileSize(INVALID_HANDLE_VALUE, ...)`.

Maybe this should be `&& self->trackfd` instead?

Reply:
They both are state checks. The behavior is consistent with Posix -- if `trackfd` is false, we get an OSError. We could raise a different error, but then we need to change the Posix code too. Is it fine to do this in this PR, or better open a separate PR?

Reply:
We should never deliberately pass `INVALID_HANDLE_VALUE` to a Win32 API, but we can in this case.

If `trackfd` is false, we should just raise an error directly, rather than relying on the OS to generate an error code for us.